cherry_picker.py

```diff
@@ -23,65 +29,86 @@ def cherry_pick(commit_sha1, branches):
         click.echo(f"Now backporting '{commit_sha1}' into '{branch}'")
 
         # git checkout -b 61e2bc7-3.5 upstream/3.5
-        cherry_pick_branch = f"{commit_sha1[:7]}-{branch}"
+        cherry_pick_branch = f"backport-{commit_sha1[:7]}-{branch}"
         cmd = f"git checkout -b {cherry_pick_branch} {upstream}/{branch}"
-        run_cmd(cmd)
+        run_cmd(cmd, dryrun=dryrun)
 
         cmd = f"git cherry-pick -x {commit_sha1}"
-        if run_cmd(cmd):
-            cmd = f"git push origin {cherry_pick_branch}"
-            if not run_cmd(cmd):
-                click.echo(f"Failed to push to origin :(")
+        if run_cmd(cmd, dryrun=dryrun):
+            cmd = f"git push {pr_remote} {cherry_pick_branch}"
+            if not run_cmd(cmd, dryrun=dryrun):
+                click.echo(f"Failed to push to {pr_remote} :(")
             else:
-                open_pr(username, branch, cherry_pick_branch)
+                open_pr(username, branch, cherry_pick_branch, dryrun=dryrun)
         else:
             click.echo(f"Failed to cherry-pick {commit_sha1} into {branch} :(")
 
         cmd = "git checkout master"
-        run_cmd(cmd)
+        run_cmd(cmd, dryrun=dryrun)
 
         cmd = f"git branch -D {cherry_pick_branch}"
-        if run_cmd(cmd):
-            click.echo(f"branch {cherry_pick_branch} has been deleted.")
+        if run_cmd(cmd, dryrun=dryrun):
+            if not dryrun:
+                click.echo(f"branch {cherry_pick_branch} has been deleted.")
         else:
             click.echo(f"branch {cherry_pick_branch} NOT deleted.")
 
 
-def get_git_upstream_remote():
+def get_git_fetch_remote():
     """Get the remote name to use for upstream branches
     Uses "upstream" if it exists, "origin" otherwise
     """
     cmd = "git remote get-url upstream"
-    if run_cmd(cmd):
+    if run_cmd(cmd, dryrun=False):
         return "upstream"
     else:
         return "origin"
 
+def get_git_push_remote():
+    """Get the remote name to use for upstream branches
+    Uses "pr" if it exists, "origin" otherwise
+    """
+    cmd = "git remote get-url pr"
+    if run_cmd(cmd, dryrun=False):
+        return "pr"
```

Comment:
I actually learned origin/pr before upstream/origin, as the expected pattern was that you'd have a read-only clone of upstream *first*, and only decide to submit PRs later.

I like the "--push" CLI option though, leaving the automated guess solely for the upstream vs origin case.